src/celestia/celx.cpp

```diff
@@ -1233,13 +1235,15 @@ const char* Celx_SafeGetString(lua_State* l,
             Celx_DoError(l, errorMsg);
         return nullptr;
     }
-    if (!lua_isstring(l, index))
+    const char *s;
+    if (!lua_isstring(l, index) || 
+        ((s = lua_tostring(l, index)) == nullptr))
     {
         if (fatalErrors & WrongType)
             Celx_DoError(l, errorMsg);
         return nullptr;
     }
-    return lua_tostring(l, index);
+    return s;
```

Comment:
please revert this change. if the object is `lua_isstring` then `lua_tostring` won't return nullptr. And you still return nullptr to the caller.

let me show a simple example:
```
#include <iostream>
#include <string>

const char* fn() {return nullptr;}

int main()
{
  std::string s = fn();
  std::cout << s << '\n';
}
```

Now compile and run it: `g++ t.cc && ./a.out`

You get:
```
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
Aborted
```

And that's why I'm telling you that Celx_SafeGetString is not so safe and you should check it's return.